src/components/Footer.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+interface FooterProps {
+  todos: Todo[];
+  filter: 'all' | 'active' | 'completed';
+  onSetFilter: (filter: 'all' | 'active' | 'completed') => void;
+  onClearCompleted: () => void;
+}
+
+export const Footer: React.FC<FooterProps> = ({
+  todos,
+  filter,
+  onSetFilter,
+  onClearCompleted,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {todos.filter(todo => !todo.completed).length} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={`filter__link ${filter === 'all' ? 'selected' : ''}`}
+          data-cy="FilterLinkAll"
+          onClick={() => onSetFilter('all')}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={`filter__link ${filter === 'active' ? 'selected' : ''}`}
+          data-cy="FilterLinkActive"
+          onClick={() => onSetFilter('active')}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={`filter__link ${filter === 'completed' ? 'selected' : ''}`}
+          data-cy="FilterLinkCompleted"
+          onClick={() => onSetFilter('completed')}
+        >
+          Completed
+        </a>
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method